src/robot.cpp

```diff
@@ -608,15 +608,29 @@ void Robot::executeManipulationPath()
       planning_groups_.find(planning_group_name);  ///< Will throw if group does not exist
   if (group_it != planning_groups_.end())
   {
-    bool success;
+    bool success = false;
     group_it->second->setStartStateToCurrentState();
     group_it->second->setRandomTarget();
-    success = static_cast<bool>(group_it->second->execute(last_plan));
+    //success = static_cast<bool>(group_it->second->execute(last_plan));
+    size_t i=0;
+    while (!success && i<3) 
```

Comment:
why not  `for (size_t i=0; !success || i<3; i++) {  ...  }`? just a bit more compact